v2/router.go

```diff
@@ -48,9 +48,37 @@ type Procedure struct {
 	Signature string
 }
 
+// EncodingProcedure specifies a single encoding-level handler and is mapped to a TransportProcedure by the router
+type EncodingProcedure struct {
+	// Name of the procedure.
+	Name string
+
+	// Service or empty to use the default service name.
+	Service string
+
+	// HandlerSpec specifying which handler and rpc type.
+	HandlerSpec EncodingHandlerSpec
+
+	// Encoding of the handler.
+	// (if present).
```

Comment:
I know this will be removed from this PR, but a note for the future: what does it mean for an `EncodingProcedure` to have an empty `Encoding`?